### PR TITLE
refactor: not allowed int64 type as time index

### PR DESCRIPTION
**Pull request by QuenKar**
I hereby agree to the terms of the [GreptimeDB CLA](https://gist.github.com/xtang/6378857777706e568c1949c7578592cc)

## What's changed and what's your intention?

_PLEASE DO NOT LEAVE THIS EMPTY !!!_

Please explain IN DETAIL what the changes are in this PR and why they are needed:

main change:
1. remove `is_timestamp_compatible()` from `datatype` trait.
2. remove `int64` => `timestamp` from match arms.
3. remove `int64` to `timestamp` from mito2.

## Checklist

- [ ]  I have written the necessary rustdoc comments.
- [ ]  I have added the necessary unit tests and integration tests.

### CODE REVIEW

File: src/store-api/src/metadata.rs [hunk 1]
```diff
@@ -356,13 +355,10 @@ impl RegionMetadata {
...
             ensure!(
-                column_metadata
-                    .column_schema
-                    .data_type
-                    .is_timestamp_compatible(),
+                column_metadata.column_schema.data_type.is_timestamp(),
                 InvalidMetaSnafu {
                     reason: format!(
-                        "{} is not timestamp compatible",
+                        "column:{} is not timestamp type",
```
Comment (killme2008) [inline src/store-api/src/metadata.rs:361]:
```suggestion
                        "column `{}` is not timestamp type",
```

File: src/store-api/src/metadata.rs [hunk 2]
```diff
@@ -676,7 +672,7 @@ mod test {
         builder.push_column_metadata(col);
         let err = builder.build().unwrap_err();
         assert!(
-            err.to_string().contains("ts is not timestamp compatible"),
+            err.to_string().contains("column:ts is not timestamp type"),
```
Comment (killme2008) [inline src/store-api/src/metadata.rs:675]:
```suggestion
            err.to_string().contains("column `ts` is not timestamp type"),
```

**Review by killme2008 — APPROVED** [review]

**Comment by killme2008** [comment]
@v0y4g3r PTAL

**Review by evenyag — APPROVED** [review]
LGTM

**Review by v0y4g3r — APPROVED** [review]
LGTM

**Pull request merged**